diff --git src/parser.rs src/parser.rs
index afb9f86..50c13d9 100644
--- src/parser.rs
+++ src/parser.rs
@@ -182,6 +182,7 @@ impl Library {
                     union_count += 1;
                 })
             }
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -305,6 +306,7 @@ impl Library {
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
             "doc-deprecated" => parser.text().map(|t| doc_deprecated = Some(t)),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -441,6 +443,7 @@ impl Library {
                 Ok(())
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -553,6 +556,7 @@ impl Library {
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
             "virtual-method" => parser.ignore_element(),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -593,6 +597,7 @@ impl Library {
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
             "doc-deprecated" => parser.text().map(|t| doc_deprecated = Some(t)),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -640,6 +645,7 @@ impl Library {
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
             "doc-deprecated" => parser.text().map(|t| doc_deprecated = Some(t)),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -709,6 +715,7 @@ impl Library {
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
             "doc-deprecated" => parser.text().map(|t| doc_deprecated = Some(t)),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -755,6 +762,7 @@ impl Library {
                 Ok(())
             }
             "doc" => parser.text().map(|t| doc = Some(t)),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
@@ -834,6 +842,7 @@ impl Library {
             "doc" => parser.text().map(|t| doc = Some(t)),
             "doc-deprecated" => parser.text().map(|t| doc_deprecated = Some(t)),
             "doc-version" => parser.ignore_element(),
+            "source-position" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
