--- NM-1.0.gir
+++ NM-1.0.gir
@@ -1530,7 +1530,6 @@
               glib:get-type="nm_checkpoint_create_flags_get_type"
               c:type="NMCheckpointCreateFlags">
       <doc xml:space="preserve">The flags for CheckpointCreate call</doc>
-      <doc-version xml:space="preserve">(gi flags generated since 1.12)</doc-version>
       <member name="none"
               value="0"
               c:identifier="NM_CHECKPOINT_CREATE_FLAG_NONE"
